orca-clouddriver/src/main/java/com/netflix/spinnaker/orca/clouddriver/tasks/monitoreddeploy/MonitoredDeployBaseTask.java

```diff
@@ -267,28 +264,24 @@ private MonitoredDeployStageData getStageContext(StageExecution stage) {
     return stage.mapTo(MonitoredDeployStageData.class);
   }
 
-  private String getRetrofitLogMessage(Response response) {
-    if (response == null) {
+  private String getErrorMessage(SpinnakerServerException spinnakerException) {
+    if (!(spinnakerException instanceof SpinnakerHttpException)) {
       return "<NO RESPONSE>";
     }
 
-    String body = "";
-    String status = "";
-    String headers = "";
+    SpinnakerHttpException httpException = (SpinnakerHttpException) spinnakerException;
 
     try {
-      status = String.format("%d (%s)", response.getStatus(), response.getReason());
-      body =
-          CharStreams.toString(
-              new InputStreamReader(response.getBody().in(), StandardCharsets.UTF_8));
-      headers =
-          response.getHeaders().stream().map(Header::toString).collect(Collectors.joining("\n"));
+      String body =
+          Optional.ofNullable(objectMapper.writeValueAsString(httpException.getResponseBody()))
+              .orElse("");
+      return String.format("headers: %s\nresponse body: %s", httpException.getHeaders(), body);
     } catch (Exception e) {
       log.error(
-          "Failed to fully parse retrofit error while reading response from deployment monitor", e);
+          "Failed to fully parse http response while reading response from deployment monitor", e);
     }
 
-    return String.format("status: %s\nheaders: %s\nresponse body: %s", status, headers, body);
+    return "headers: \nresponse body: ";
```

Comment:
I think we still need String.format here, to include the actual headers and response body.

Reply:
I dont understand what is the need to format the String here. If you see the code before the PR, 
We had the initialisation like below:
```
    String body = "";
    String headers = "";
```

In case of any exceptions while parsing the response details the default empty values are returned, which now I have simplified and doing the same thing.

The actual headers and body are parsed and returned inside the try block itself in case of success scenario.

Reply:
Aaah, I see.  I think it'd be better to indicate some kind of error than possibly misleading people into thinking there were no headers/body.

Reply:
The indication about the error will be printed in the log statement in the above catch block : https://github.com/spinnaker/orca/blob/e21a5cc06971cddde34d9b3ac59264b7aa31ea86/orca-clouddriver/src/main/java/com/netflix/spinnaker/orca/clouddriver/tasks/monitoreddeploy/MonitoredDeployBaseTask.java#L279C7-L281C6

This should indicate the users on why the headers and response bodies are empty